src/main/java/edu/harvard/iq/dataverse/flyway/StartupFlywayMigrator.java

```diff
@@ -0,0 +1,28 @@
+package edu.harvard.iq.dataverse.flyway;
+
+import org.flywaydb.core.Flyway;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.Resource;
+import javax.ejb.Singleton;
+import javax.ejb.Startup;
+import javax.sql.DataSource;
+
+@Startup
+@Singleton
+class StartupFlywayMigrator {
+
+    @Resource(lookup = "jdbc/VDCNetDS")
+    private DataSource dataSource;
+
+    @PostConstruct
+    void migrateDatabase() {
+
```

Comment:
IMHO a check on `dataSource == null` should be done here or a `try/catch` for NPEs, so the user gets a reasonable error message. That should finally result in an `EJBException` getting thrown, so deployment fails.